src/mesh/mesh.hpp

```diff
@@ -112,13 +113,56 @@ class MeshBlock {
   LogicalLocation loc;
   RegionSize block_size;
   // for convenience: "max" # of real+ghost cells along each dir for allocating "standard"
-  // sized MeshBlock arrays, depending on ndim (i.e. ncells2=nx2+2*NGHOST if nx2>1)
-  int ncells1, ncells2, ncells3;
-  // on 1x coarser level MeshBlock (i.e. ncc2=nx2/2 + 2*NGHOST, if nx2>1)
-  int ncc1, ncc2, ncc3;
-  int is, ie, js, je, ks, ke;
+  // sized MeshBlock arrays, depending on ndim i.e.
+  //
+  // cells.nx2 =    nx2      + 2*NGHOST if   nx2 > 1
+  // (entire)    (interior)               (interior)
+  //  
+  // Assuming we have a block cells, and nx2 = 6, and NGHOST = 1
+  //
+  // <----- nx1 = 8 ---->
+  //       (entire)
+  //
+  //     <- nx1 = 6 ->
+  //       (interior)
+  //
+  //  - - - - - - - - - -   ^
+  //  |  |  ghost    |  |   | 
+  //  - - - - - - - - - -   |         ^
+  //  |  |     ^     |  |   |         |
+  //  |  |     |     |  |  nx2 = 8    nx2 = 6 
+  //  |  | interior  |  | (entire)   (interior)
+  //  |  |     |     |  |             |
+  //  |  |     v     |  |   |         v 
+  //  - - - - - - - - - -   |
+  //  |  |           |  |   | 
+  //  - - - - - - - - - -   v 
+  //
+  IndexShape cells;
```

Comment:
I think naming this `cells` is a little confusing, since the meshblocks do in fact have cells and these aren't they. Maybe `cellbounds` or something similar?

Reply:
`domain` also wouldn't be bad.